contributors/design-proposals/mount-options.md

```diff
@@ -35,14 +57,37 @@ Mount options can be specified as `mount-options` annotations in PV. For example
       server: 172.17.0.2
 ```
 
+### Mount option support in Storage Classes
+
+Kubernetes admin can also specify mount option as a parameter in storage class.
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: slow
+provisioner: kubernetes.io/glusterfs
+parameters:
+  type: gp2
+  mountOptions: "auto_mount"
+```
+
+The mount option specified in Storage Class will be used while provisioning persistent volumes
+and added as a field to PVs.
+
+If admin has configured mount option for a storage type that does not support mount options,
+then a "provisioning failed" event will be added to PVC and PVC will stay in pending state.
+
+Also, if configured mount option is invalid then corresponding mount time failure error will be added to pod object.
+
 
 ## Preventing users from specifying mount options in inline volume specs of Pod
 
 While mount options enable more flexibility in how volumes are mounted, it can result
 in user specifying options that are not supported or are known to be problematic when
 using inline volume specs.
 
-After much deliberation it was decided that - `mount-options` as an API parameter will not be supported
+After much deliberation it was decided that - `mountOptions` as an API parameter will not be supported
 for inline volume specs.
```

Comment:
Thanks a lot for the link! I guess I'm with saad-ali, but all makes sense and it's nice to see the background. Thanks :)